core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -340,19 +340,20 @@ class LogTest extends JUnitSuite {
       // first test a log segment starting at 0
       val logProps = new Properties()
       logProps.put(LogConfig.SegmentBytesProp, 100: java.lang.Integer)
+      logProps.put(LogConfig.RetentionMsProp, 0: java.lang.Integer)
       val log = new Log(logDir, LogConfig(logProps), recoveryPoint = 0L, time.scheduler, time = time)
       for(i <- 0 until messagesToAppend)
-        log.append(TestUtils.singleMessageSet(i.toString.getBytes))
+        log.append(TestUtils.singleMessageSet(payload = i.toString.getBytes, timestamp = time.milliseconds - 10))
```

Comment:
This test was having random failures as the predicate `startMs - _.largestTimestamp > config.retentionMs` would fail. The timestamp needs to be in the past.